### ISSUE TITLE
Incompatibility with Cobra's PersistentFlags

### ISSUE THREAD

**Issue by Nivl** [cornfeedhobo/pflag#22]
It seems that Cobra's `PersistentFlags()` doesn't work with this lib:

Code:

```
package main

import (
	"fmt"

	"github.com/spf13/cobra"
)

type config struct {
	C string
}

func newRootCmd() *cobra.Command {
	cmd := &cobra.Command{
		Use: "command",
	}

	cfg := &config{}
	cmd.PersistentFlags().StringVarS(&cfg.C, "C", "C", "", "help")

	cmd.AddCommand(newHelloWorldCmd())

	return cmd
}

func newHelloWorldCmd() *cobra.Command {
	cmd := &cobra.Command{
		Use: "hello",
	}

	cmd.RunE = func(cmd *cobra.Command, args []string) error {
		fmt.Println("world")
		return nil
	}
	return cmd
}

func main() {
	newRootCmd().Execute()
}
```

The `help` works as intended and show the flag:

```
❯ cmd help
Usage:
  command [command]

Available Commands:
  hello
  help        Help about any command

Flags:
  -C string    help
  -h, --help   help for command

Use "command [command] --help" for more information about a command.
```

But the flag is not found:

```
❯ cmd -C something hello
Error: unknown shorthand flag: 'C' in -C
```

**Comment by cornfeedhobo** [comment]
@Nivl Thanks so much for your report! I can reproduce this locally. I will do my best to find time for investigation. If anyone else gets to this, I'll be sure to be responsive on the reviews.

The PR that fixes this should also fix #16

**Comment by cornfeedhobo** [comment]
@Nivl I've identified the offending commit and code. Working on a fix.

**Comment by cornfeedhobo** [comment]
@Nivl Thank you for your patience and your report! This led to me figuring out cobra integration testing and then uncovering a number of other bugs along the way. In response, I have moved the `master` branch off to a dedicated branch and published a re-fork of the merged down result to `main` and a new `v1.1.0` release. Please give it a try and let me know if you have any troubles.

I'm going to close this for now, but feel free to re-open if you still find issues.

**Comment by Nivl** [comment]
Just tested, it works perfectly! Thanks @cornfeedhobo
